src/IssueViz.Security.UnitTests/OpenInIDE/Api/OpenInIDERequestHandlerTests.cs

```diff
@@ -19,72 +19,251 @@
         }
 
         [TestMethod]
         public async Task ShowHotspot_InvalidArg_Throws()
         {
-            IOpenInIDERequestHandler testSubject = new OpenInIDERequestHandler(Mock.Of<IOpenInIDEStateValidator>(), Mock.Of<ISonarQubeService>(), Mock.Of<ILogger>());
-
             Func<Task> act = async () => await testSubject.ShowHotspotAsync(null);
             act.Should().ThrowExactly<ArgumentNullException>().And.ParamName.Should().Be("request");
         }
 
         [TestMethod]
-        public async Task ShowHotpot_IdeNotInCorrectState_NoActionTaken()
+        public async Task ShowHotpot_IdeNotInCorrectState_NoFurtherProcessing()
         {
-            var validatorMock = CreateValidator("http://localhost/", "project", "org", false);
-            var request = new TestShowHotspotRequest
-            {
-                ServerUrl = new Uri("http://localhost/"),
-                ProjectKey = "project",
-                HotspotKey = "any",
-                OrganizationKey = "org"
-            };
-            var serverMock = new Mock<ISonarQubeService>();
+            InitializeStateValidator(ValidRequest, false);
 
-            IOpenInIDERequestHandler testSubject = new OpenInIDERequestHandler(validatorMock.Object, serverMock.Object, Mock.Of<ILogger>());
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
 
-            await testSubject.ShowHotspotAsync(request);
+            CheckInfoBarCleared();
+            CheckInfoBarShown();
+            CheckCalled(toolWindowService, stateValidatorMock);
+            CheckNotCalled(serverMock, converterMock, navigatorMock, storeMock);
+        }
+
+        [TestMethod]
+        public async Task ShowHotspot_FailedToRetrieveHotspotData_NoFurtherProcessing()
+        {
+            InitializeStateValidator(ValidRequest, true);
+            SetServerResponse(ValidRequest, null);
 
-            validatorMock.VerifyAll();
-            serverMock.Invocations.Should().BeEmpty();
+            // Act
+            await testSubject.ShowHotspotAsync(ValidRequest)
+                .ConfigureAwait(false);
+
+            CheckInfoBarCleared();
+            CheckInfoBarShown();
+            CheckCalled(toolWindowService, stateValidatorMock, serverMock);
+            CheckNotCalled(converterMock, navigatorMock, storeMock);
         }
 
-        private static Mock<IOpenInIDEStateValidator> CreateValidator(string uri, string projectKey, string orgKey, bool canHandleRequest)
+        [TestMethod]
+        public async Task ShowHotspot_ConversionFailed_NoFurtherProcessing()
         {
-            var mockValidator = new Mock<IOpenInIDEStateValidator>();
-            mockValidator.Setup(x => x.CanHandleOpenInIDERequest(It.Is<Uri>(x => x.ToString() == uri), projectKey, orgKey))
+            InitializeStateValidator(ValidRequest, true);
+            SetServerResponse(ValidRequest, ValidServerHotspot);
```

Comment:
I guess mock-wise it's redundant to say
`SetConversionResponse(ValidServerHotspot, null);`  but I think it's worth adding anyway, for readability.